src/internal/m365/collection/drive/collections.go

```diff
@@ -866,19 +866,25 @@ func (c *Collections) processItem(
 		// This check is to ensure that if a folder was deleted and
 		// recreated multiple times between a backup, we only use the
 		// final one.
-		pid, found := seenFolders.IDOf(collectionPath.String())
-
-		recreated := found && pid != itemID
-		if recreated {
+		alreadyHandledFolderID, collPathAlreadyExists := seenFolders[collectionPath.String()]
+		collPathAlreadyExists = collPathAlreadyExists && alreadyHandledFolderID != itemID
+
+		if collPathAlreadyExists {
+			// we don't have a good way of juggling multiple previous paths
+			// at this time.  If a path was declared twice, it's a bit ambiguous
+			// which prior data the current folder now contains.  Safest thing to
+			// do is to call it a new folder and ingest items fresh.
```

Comment:
is this safe in all cases? Does moving a folder to the spot of the deleted folder also lead to a missing delete record in the delta result? If it does then we could end up duplicating (part of) the moved folder's items and never actually put the folder in the proper place in the kopia hierarchy

Reply:
also I think we need to explore what happens if we don't see a delta record for a deleted folder that had children. If we also don't see delta records showing the deletion of the child folders/items then the previous paths map will get out of sync